static/js/ui_util.ts

```diff
@@ -47,6 +47,37 @@ export function update_unread_count_in_dom($unread_count_elem: JQuery, count: nu
     $unread_count_span.text(count);
 }
 
+export function mask_unread_count_in_dom($unread_count_elem: JQuery, count: number): void {
+    const $unread_count_span = $unread_count_elem.find(".unread_count");
+    $unread_count_span.text(count);
+    $unread_count_span.hide();
+
+    const $masked_count_span = $unread_count_elem.find(".masked_unread_count");
+    $masked_count_span.show();
+
+    // Attach event listener to show the unread count on hover.
+    $unread_count_elem.on("mouseenter", () => {
+        $unread_count_span.show();
+        $masked_count_span.hide();
+    });
+    $unread_count_elem.on("mouseleave", () => {
+        $unread_count_span.hide();
+        $masked_count_span.show();
+    });
+}
```

Comment:
The frontend logic here needs refactoring; it feels like you have a hundred semi-duplicate `.save()` / `.hide()` calls here.

I also think you can do all of this with CSS, and it'll be a lot nicer. For example, the following would do the toggle-on-hover behavior more reliably than the `mouseenter/mouseleave` handlers:

```
.narrow-filter .out_of_home_view {
    .masked_unread_count {
        display: inline;
     }
   .unread_count {
        display: none;
    }
}

.narrow-filter:hover .out_of_home_view {
    .masked_unread_count {
        display: none;
     }
   .unread_count {
        display: inline;
    }
}
```

You'll need to set a new outer CSS class when rendering rows to communicate the value of the new display display to the CSS logic -- so probably `out_of_home_view` wouldn't actually appear in the CSS, instead being something like `unread_counts_masked` that would be computed in template rendering.